io/shared/src/main/scala/fs2/io/process/Process.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2
+package io
+package process
+
+sealed trait Process[F[_]] {
+
+  def isAlive: F[Boolean]
+
+  def exitValue: F[Int]
+
+  def stdin: Pipe[F, Byte, Nothing]
+
+  def stdout: Stream[F, Byte]
+
+  def stderr: Stream[F, Byte]
+
+}
```

Comment:
This API definitely needs some thinking about. Particularly about semantics of `stdin`, `stdout`, and `stderr`. Can you compile these streams at most once, or many times? Do they close the underlying streams when complete? Along these lines, do we want APIs to read/write a `Chunk` directly?

This is all possible I believe but trickier to support. Easiest of course is compile-at-most-once.

Reply:
> Easiest of course is compile-at-most-once. 

This matches my personal intuition

Reply:
I'm also curious about how piping processes looks like

Reply:
> I'm also curious about how piping processes looks like

Not a lot of fun right now. I wonder if we added a `Monoid[Process]` for this.

Reply:
> Monoid[Process] 

Interesting idea, though what'd happen to the stderr ? :/

Reply:
I guess it would be similar to piping in the shell, only stdout. Also as suggested in https://github.com/typelevel/fs2/pull/3149#issuecomment-1445207210 we could add a feature to redirect stderr to stdout.

On second thought, less sure about `Monoid` (what's the identity process 🤔) but `Semigroup[Process]` seems plausible.